File: dalme_ui/src/schemas/countries.js
import * as yup from "yup";

export const countryListSchema = yup.array().of(
  yup
    .object()
    .shape({
      id: yup.number().required(),
      name: yup.string().required(),
      alpha2Code: yup.string().required(),
      alpha3Code: yup.string().required(),
      numCode: yup.number().required(),
    })
    .camelCase(),
);

File: dalme_api/serializers/countries.py
from dalme_app.models import CountryReference

from .base_classes import DynamicSerializer


class CountryReferenceSerializer(DynamicSerializer):
    """Serializer for countries."""

    class Meta:  # noqa: D106
        model = CountryReference
        fields = ('id', 'name', 'alpha_2_code', 'alpha_3_code', 'num_code')
        field_sets = {
            'option': ['id', 'name'],
            'attribute': ['id', 'name'],
        }
